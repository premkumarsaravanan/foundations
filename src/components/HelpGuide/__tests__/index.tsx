import React, { useState } from 'react'
import { mount, shallow } from 'enzyme'
import { HelpGuide } from '..'

const HelpGuideWrapper = () => {
  return (
    <div style={{ padding: 20 }}>
      <HelpGuide>
        <HelpGuide.Step id="step-1" component={() => <div className="step-1">Step 1</div>}></HelpGuide.Step>
        <HelpGuide.Step id="step-2" component={() => <div className="step-2">Step 2</div>}></HelpGuide.Step>
        <HelpGuide.Step id="step-3" component={() => <div className="step-3">Step 3</div>}></HelpGuide.Step>
        <HelpGuide.Step id="step-4" component={() => <div className="step-4">Step 4</div>}></HelpGuide.Step>
      </HelpGuide>
    </div>
  )
}

describe('HelpGuide', () => {
  it('should match a snapshot', () => {
    expect(shallow(<HelpGuideWrapper />)).toMatchSnapshot()
  })

  it('should match a snapshot', () => {
    expect(
      shallow(<HelpGuide.Step id="step-3" component={() => <div className="step-3">Step 3</div>} />)
    ).toMatchSnapshot()
  })

  afterEach(() => {
    jest.resetAllMocks()
  })
})
