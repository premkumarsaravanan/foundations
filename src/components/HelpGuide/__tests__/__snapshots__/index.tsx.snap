// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpGuide should match a snapshot 1`] = `
<div
  style={
    Object {
      "padding": 20,
    }
  }
>
  <Component>
    <Component
      component={[Function]}
      id="step-1"
    />
    <Component
      component={[Function]}
      id="step-2"
    />
    <Component
      component={[Function]}
      id="step-3"
    />
    <Component
      component={[Function]}
      id="step-4"
    />
  </Component>
</div>
`;

exports[`HelpGuide should match a snapshot 2`] = `
<Component
  className="items-center justify-between"
>
  <div
    className="helpguide-content"
  >
    <Component />
    <Component />
    <component />
  </div>
  <div
    className="helpguide-wrapper-graphic"
  >
    <div
      className="helpguide-graphic"
    />
  </div>
</Component>
`;
